apps/04TowersOfHanoi.js

```diff
@@ -19,24 +19,48 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
   // Your code here
-
+  var start = stacks[startStack].pop();
+  stacks[endStack].push(start);
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
+var legal = false
+var startBlock = stacks[startStack][stacks[startStack].length -1]
+var endBlock = stacks[endStack][stacks[endStack].length - 1]
 
+if(stacks[endStack].length >=0){
+  legal = true;
 }
 
-function checkForWin() {
-  // Your code here
+if(startBlock < endBlock){
+  legal = true;
+}
 
+if(startBlock > endBlock){
```

Comment:
This looks like a good opportunity for an `else` block attached to the previous `if`